### DISCUSSION TITLE
Spacy v3 example object for scorer function

### DISCUSSION THREAD

**Discussion by albin02t** [explosion/spaCy#8897] [Help: Coding & Implementations] [answered]
Hi everyone, 
I wanted to run the scorer class to evaluate my custom NER trained model on a validation dataset.

the dataset looks like this:
`Valid_data=
    [
        "Mr. and Mrs. Dursley, of number four, Privet Drive, were proud to say that they were perfectly normal, thank you very much. They were the last people you'd expect to be involved in anything strange or mysterious, because they just didn't hold with such nonsense.",
        {
            "entities": [
                [
                    0,
                    20,
                    "PERSON"
                ]
            ]
        }
    ]]`

To apply the scorer class, I have applied the following code:
```
from spacy.tokens import Doc
from spacy.training import Example
from spacy.scorer import Scorer

for text, annots in Valid_data:
    examples=Example(nlp.make_doc(text), annots)
    pred_value=nlp(text)
    scorer.score(pred_value,examples)
print(scorer.scores)
```

![image](https://user-images.githubusercontent.com/69189799/128506742-d5e32de6-cb0e-4bcc-9424-3a081f49059d.png)

but it gives the error : 
![image](https://user-images.githubusercontent.com/69189799/128521228-2489d8ec-fe0a-492d-87b7-201759ea2b71.png)

Can someone pls help in creating the example objects for evaluation?

**Comment by svlandeg** [comment, marked as answer]
Hi! 

You can find the API for the `Example` class here: https://spacy.io/api/example

As you can see, and as the error message suggests, the constructor needs two `Example` objects.

What you're looking for is probably just `Example.from_dict` (as in the comment of your code)?
```
example=Example.from_dict(nlp.make_doc(text), annots)
```
You'll probably also want to construct the examples first, then run the evaluation on all of them?

```
examples = []
for text, annots in Valid_data:
    predicted=nlp(text)
    example=Example.from_dict(predicted, annots)
    examples.append(example)
scorer.score(examples)
```

**Reply by albin02t** [reply]
Thanks a lot ! it finally worked!! 
Also could you show the method if I want to apply it using just Example instead of Example.from_dict

**Reply by svlandeg** [reply]
If you want to use the `Example` constructor, you need to construct all the data on the `Doc` level instead of working with the annotations dictionary. Specifically for NER, you'd need to set `doc.ents` with the correct gold entities.

**Reply by albin02t** [reply]
> If you want to use the `Example` constructor, you need to construct all the data on the `Doc` level instead of working with the annotations dictionary. Specifically for NER, you'd need to set `doc.ents` with the correct gold entities.

ohkay noted. could you show me an example if possible? 
Couldn't find any practical implementations anywhere. Would be really helpful!
